### PR TITLE
[Bugfix] Fix Issues for preliminary results

### LINKED ISSUE #3333: Misleading Response/Status for Edge Case (0 of 0 Points)
### Describe the bug

Recently we created a programming exercise that had only test cases for "after due date".
Unfortunately, we got student reports because of the Artemis response for the submitted solutions.
This response is shown in the following picture in the screenshot section.
 
It looks like the mechanisms that decide whether a submission is displayed as "green" or "red" choose to be red in the edge case of "0 out of 0 points". In this edge case, however, the result should be green as long as the code compiles at least. Otherwise it will confuse the students. Another case is described in https://github.com/ls1intum/Artemis/issues/3333#issuecomment-830413583 .

After @krusche and I talked, we considered the following:

1. Preliminary must always be displayed (also in zero points case --> bugfix)
2. Preliminary must always be clickable (so that you can see more information in the details window, the explanation for preliminary should be visible here immediately, without extra hover)
3. Change color of preliminary to grey (like in the instructions)
4. Change icon of preliminary from (X) to (?)

#### To Reproduce

1. Create a programming exercise **with manual assessment and static code analysis 0% penalty** and set all test cases (in grading) to after due date. 
2. Create a compilable (or even the correct) solution
3. See the red X

#### Expected behavior

@krusche and I talked and decided: 
1. Preliminary must always be displayed (also in zero points case --> bugfix)
2. Preliminary must always be clickable (so that you can see more information in the details window, the explanation for preliminary should be visible here immediately, without extra hover)
3. Change color of preliminary to grey (like in the instructions)
4. Change icon of preliminary from (X) to (?)

#### Screenshots

0 of 0 Points:
image

1 of 1 Test ok (but not 100% Points) (should also be green from our point of view)
image

### Environment

Artemis 4.11.8

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Resolves #3333

### Description

This PR adds the "(preliminary)" text for results that are preliminary. Furthermore, preliminary results are marked in a grey color and have a question mark icon.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Java Programming Exercise. Make sure you set the dates to the future (end date and assessment dates) and select the static analysis tool with a 0% penalty
4. Click Edit in Editor and then click on Create Assignment Repository
5. Navigate to the assignment repository, edit BubbleSort.java and add `Collections.sort(input);`, and submit the code
6. The result should be colored in grey and include the "(preliminary)" text
7. Re-add the semicolon, submit the code, and navigate to "Grading".
8. Set the visibility of each test to "after due date", press "Save", and "Trigger All".
9. Navigate back to the code editor, select the assignment, and make sure the text is colored in grey and includes the "preliminary" string 
10. Click on the result and make sure the modal displayed below opens up.

### Screenshots

The visibility of all tests are set to "after due date" and the build compiles:
image

Clicking on the preliminary result (all tests are marked as "after due date") opens up a modal dialog:
image

The visibility of a single test is set to "Always" and the code implements it:
image

If the build fails, the text is red instead of grey:
image

### CODE REVIEW

**Review by dfuchss — DISMISSED** [review]
Tested on our test systems. Thanks for implementing 👍

**Review by FuchsDominik — APPROVED** [review]
Code looks fine

**Review by dfuchss — APPROVED** [review]
Re-Approve

**Review by MerlinMehmed — APPROVED** [review]
Tested on TS2, works as expected.

**Review by ndricimrr — APPROVED** [review]
Tested on TS2, working as expected!

**Review by imstreetspirit — APPROVED** [review]
code looks good to me

**Review by coolchock — APPROVED** [review]
tested on ts2 in the testing session, works as described

File: src/main/webapp/app/exercises/shared/result/result.component.html [hunk 1]
```diff
@@ -18,17 +18,24 @@
...
                         {{ resultString }}
                     </span>
                 </span>
-                <span *ngIf="!hasFeedback && onlyShowSuccessfulCompileStatus" [ngbTooltip]="resultTooltip" container="body">
-                    {{ resultString }}
-                </span>
-                <span *ngIf="!hasFeedback && !onlyShowSuccessfulCompileStatus" [ngbTooltip]="resultTooltip" container="body">
-                    {{ result.resultString }}
-                </span>
+                <ng-container *ngIf="!resultIsPreliminary(result)">
```
Comment (nicolasruscher) [inline src/main/webapp/app/exercises/shared/result/result.component.html:26]: I mean function calls in html are to be omitted, but I think in this context the resulting html would become kinda ugly, so I guess its ok here

**Review by nicolasruscher — APPROVED** [review]
Works! 👍 The small comment is just a minor thing

**Review by KiraNickel — APPROVED** [review]
Tested on ts3, works as described 👍🏼

**Pull request merged**